code/API_definitions/kyc-tenure.yaml

```diff
@@ -0,0 +1,485 @@
+              description: Resource is not found
+              value:
+                status: 404
+                code: NOT_FOUND
+                message: The specified resource is not found.
+            GENERIC_404_DEVICE_NOT_FOUND:
+              description: Device identifier not found
+              value:
+                status: 404
+                code: DEVICE_NOT_FOUND
+                message: Device identifier not found.
+
+    Generic406:
+      description: Not Acceptable
+      headers:
+        x-correlator:
+          $ref: "#/components/headers/X-Correlator"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorInfo"
+          examples:
+            GENERIC_406_NOT_ACCEPTABLE:
+              description: API Server does not accept the media type (`Accept-*` header) indicated by API client
+              value:
+                status: 406
+                code: NOT_ACCEPTABLE
+                message: The server cannot produce a response matching the content requested by the client through `Accept-*` headers.
+
+    Generic415:
+      description: Unsupported Media Type
+      headers:
+        x-correlator:
+          $ref: "#/components/headers/X-Correlator"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorInfo"
+          examples:
+            GENERIC_415_UNSUPPORTED_MEDIA_TYPE:
+              description: Payload format of the request is in an unsupported format by the Server. Should not happen
+              value:
+                status: 415
+                code: UNSUPPORTED_MEDIA_TYPE
+                message: The server refuses to accept the request because the payload format is in an unsupported format.
+
+    Generic422:
+      description: Unprocessable Content
+      headers:
+        x-correlator:
+          $ref: "#/components/headers/X-Correlator"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorInfo"
+          examples:
+            GENERIC_422_DEVICE_IDENTIFIERS_MISMATCH:
+              description: Inconsistency between device identifiers not pointing to the same device
+              value:
+                status: 422
+                code: DEVICE_IDENTIFIERS_MISMATCH
+                message: Provided device identifiers are not consistent.
+            GENERIC_422_DEVICE_NOT_APPLICABLE:
+              description: Service is not available for the provided device
+              value:
+                status: 422
+                code: DEVICE_NOT_APPLICABLE
+                message: The service is not available for the provided device.
+            GENERIC_422_UNIDENTIFIABLE_DEVICE:
+              description: No identifier provided for any device
+              value:
+                status: 422
+                code: UNIDENTIFIABLE_DEVICE
+                message: No identifier provided for any device
```

Comment:
Same as the previous comment. Here the device object is not used.

1. DEVICE_IDENTIFIERS_MISMATCH -> since there are no multiple identifiers in the input this one makes no sense here. It is used, for example, when providing an ip and a phone number from different devices.
2. DEVICE_NOT_APPLICABLE -> in Sim swap we defined 422 `NOT_SUPPORTED` to indicate that the service is not supported for the provided phone number
3. UNIDENTIFIABLE_DEVICE -> Adapted to `UNIDENTIFIABLE_PHONE_NUMBER` with the same meaning but for APIs that use the phone number as input

references for the errors discussions:
https://github.com/camaraproject/SimSwap/issues/117
https://github.com/camaraproject/Commonalities/pull/145#issuecomment-2127428748

Reply:
`phoneNumber` is a subset of the `Device` object, and I don't think special rules should apply just because we propose not to allow device subscriptions to be identified by IP address or network access identifier.

But also, and this is important, a 3-legged token **can** be generated from the IP address. So just because we do not **explicitly** allow IP address as a device subscription identifier does not mean that IP address cannot be used to generate a 3-legged token.

More specifically:
- `DEVICE_IDENTIFIERS_MISMATCH` is intended when the "device" identified by a 3-legged token is different to that explicitly passed (in this case, by `phoneNumber`). There are privacy issues here if we confirm whether or not the identifiers match, so this one may yet be removed by Commonalities.
- `DEVICE_NOT_APPLICABLE` is what SIM Swap should have used rather than define their own error code
- `UNIDENTIFIABLE_DEVICE` is when a 2-legged token is used with no `phoneNumber` being provided.

I don't like that APIs define new error codes without discussing this in Commonalities and getting them standardised.

Reply:
`DEVICE_IDENTIFIERS_MISMATCH` is redundant because of the privacy issues, so I removed it

Reply:
See [this](https://github.com/camaraproject/Commonalities/issues/321#issuecomment-2429158445) comment. Proposal would be to use `SERVICE_NOT_APPLICABLE` and `MISSING_IDENTIFIER`.